src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/representation/token/TokenTypeImpl.kt

```diff
@@ -0,0 +1,14 @@
+package org.danilopianini.plagiarismdetector.analyzer.representation.token
+
+import kotlinx.serialization.Serializable
+
+/**
+ * A simple class implementing a [TokenType].
+ */
+@Serializable
+data class TokenTypeImpl(
+    override val name: String,
+    override val languageConstructs: Collection<String>
```

Comment:
```suggestion
    override val languageConstructs: Iterable<String>,
```

Reply:
I used a collection because I load the token types from a yaml configuration file.

Reply:
Question: can the constructs appear twice, or would that be an error?

Reply:
Would be an error... better a Set.